File: Algorithms/p142_Linked_List_Cycle_II/p142_Linked_List_Cycle_II_testcase.cpp
int main(){
    ListNode* t1 = new ListNode(1);
    ListNode* t2 = new ListNode(2);
    ListNode* t3 = new ListNode(3);
    ListNode* t4 = new ListNode(4);
    ListNode* t5 = new ListNode(5);
    t1->next = t2;
    t2->next = t3;
    t3->next = t4;
    t4->next = t5;
    t5->next = t3;

    Solution s;
    ListNode* res0 = s.detectCycle(0);
    ListNode* res1 = s.detectCycle(t1);
    cout<<(res0?res0->val:0)<<endl;
    cout<<(res1?res1->val:0)<<endl;

    return 0;
}

File: Algorithms/p142_Linked_List_Cycle_II/p142_Linked_List_Cycle_II_testcase.py
t1 = ListNode(1)
t2 = ListNode(2)
t3 = ListNode(3)
t4 = ListNode(4)
t5 = ListNode(5)
t1.next = t2
t2.next = t3
t3.next = t4
t4.next = t5
t5.next = t3

s = Solution()
res0 = s.detectCycle(None)
res1 = s.detectCycle(t1)
print(res0.val if res0 else 0)
print(res1.val if res1 else 0)
